proxy/src/transport/tls/config.rs

```diff
@@ -0,0 +1,142 @@
+use std::{
+    fs::File,
+    io::{self, Cursor, Read},
+    path::PathBuf,
+    sync::Arc,
+};
+
+use super::{
+    cert_resolver::CertResolver,
+
+    rustls,
+    untrusted,
+    webpki,
+};
+
+/// Not-yet-validated settings that are used for both TLS clients and TLS
+/// servers.
+///
+/// The trust anchors are stored in PEM format because, in Kubernetes, they are
+/// stored in a ConfigMap, and until very recently Kubernetes cannot store
+/// binary data in ConfigMaps. Also, PEM is the most interoperable way to
+/// distribute trust anchors, especially if it is desired to support multiple
+/// trust anchors at once.
+///
+/// The end-entity certificate and private key are in DER format because they
+/// are stored in the secret store where space utilization is a concern, and
+/// because PEM doesn't offer any advantages.
+#[derive(Debug)]
+pub struct CommonSettings {
+    /// The trust anchors as concatenated PEM-encoded X.509 certificates.
+    pub trust_anchors: PathBuf,
+
+    /// The end-entity certificate as a DER-encoded X.509 certificate.
+    pub end_entity_cert: PathBuf,
+
+    /// The private key in DER-encoded PKCS#8 form.
+    pub private_key: PathBuf,
+}
+
+/// Validated configuration common between TLS clients and TLS servers.
+pub struct CommonConfig {
+    cert_resolver: Arc<CertResolver>,
+}
+
+/// Validated configuration for TLS servers.
+#[derive(Clone)]
+pub struct ServerConfig(pub(super) Arc<rustls::ServerConfig>);
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
would `impl Error` change anything in the current behavior? if not, i don't think we should worry about this until we actually use the implementation.

Reply:
Agreed, but I think we ought to eventually make a follow-up change to add user-facing formatting for the error messages.

Reply:
> Agreed, but I think we ought to eventually make a follow-up change to add user-facing formatting for the error messages.

I think it would be better to hold off on any work like that in general until we come up with an overall design philosophy around this, in general, for Conduit.

Reply:
> I think it would be better to hold off on any work like that in general until we come up with an overall design philosophy around this, in general, for Conduit.

I agree that we ought to do that. However, I will add that a wide majority (possibly all?) of the error types we log at the `ERROR` level are formatted with `fmt::Display` rather than `Debug`.